### PR TITLE
feat:  Expose Soft Pause Endpoint 

### LINKED ISSUE #16874: Expose the Soft Pause endpoint in the Admin API in the Broker
Expose the soft pause endpoint in the admin API of the broker, similar to the wai we have with the pause endpoint.

### LINKED ISSUE #17256: Expose endpoint in the Gateway
Expose the soft pause endpoint in the admin API of the Gateway , similar to the way we have with the pause endpoint.

### PR DESCRIPTION
**Pull request by rodrigo-lourenco-lopes**
## Description

This pull request exposes the soft pause functionality both at the broker level as at the gateway level.

The new endpoint added for the softpause functionality reuses the pause by just adding the parameter "soft" set as true. endpoint `/actuator/exporting/pause?soft=true` .

## Related issues

closes https://github.com/camunda/zeebe/issues/16874 
closes https://github.com/camunda/zeebe/issues/17256

### CODE REVIEW

File: dist/src/main/java/io/camunda/zeebe/shared/management/ExportingEndpoint.java [hunk 1]
```diff
@@ -21,6 +21,7 @@
 @WebEndpoint(id = "exporting")
 public final class ExportingEndpoint {
   static final String PAUSE = "pause";
+  static final String SOFT_PAUSE = "softPause";
```
Comment (deepthidevaki) [inline dist/src/main/java/io/camunda/zeebe/shared/management/ExportingEndpoint.java:24]: :x: I was thinking the endpoint should be something like `exporting/pause?soft=true`. What do you think?

File: zeebe/protocol/src/main/resources/cluster-management-protocol.xml [hunk 2]
```diff
@@ -10,8 +10,9 @@
     <enum name="AdminRequestType" encodingType="uint8">
       <validValue name="STEP_DOWN_IF_NOT_PRIMARY">0</validValue>
       <validValue name="PAUSE_EXPORTING">1</validValue>
-      <validValue name="RESUME_EXPORTING">2</validValue>
-      <validValue name="BAN_INSTANCE">3</validValue>
+      <validValue name="SOFT_PAUSE_EXPORTING">2</validValue>
+      <validValue name="RESUME_EXPORTING">3</validValue>
+      <validValue name="BAN_INSTANCE">4</validValue>
```
Comment (deepthidevaki) [inline zeebe/protocol/src/main/resources/cluster-management-protocol.xml:15]: :x: This is breaking change. You cannot change the order of existing enums in the protocol. You have to add `SOFT_PAUSE_EXPORTING` at the end.

File: zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ExportingEndpointIT.java [hunk 3]
```diff
@@ -193,4 +206,181 @@ private void allPartitionsExporting() {
...
+          .query()
+          .forEach(
+              (partitionId, partitionStatus) -> {
+                secondExportedPositionPerPartition.put(
+                    partitionId, partitionStatus.exportedPosition());
+              });
+    }
+
+    assertThat(recordsAfterSoftPause).isGreaterThan(recordsBeforePause);
+    assertThat(exportedPositionPerPartition.equals(secondExportedPositionPerPartition)).isTrue();
```
Comment (deepthidevaki) [inline zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ExportingEndpointIT.java:268]:
```suggestion
    assertThat(exportedPositionPerPartition).isEqualTo(secondExportedPositionPerPartition);
```

File: zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ExportingEndpointIT.java [hunk 4]
```diff
@@ -193,4 +206,181 @@ private void allPartitionsExporting() {
...
+    assertTrue(
+        exportedPositionPerPartition.entrySet().stream()
+            .anyMatch(
+                exportedPosition ->
+                    secondExportedPositionPerPartition.get(exportedPosition.getKey())
+                        > exportedPosition.getValue()));
+  }
+
+  @Test
+  void shouldStaySoftPausedAfterRestart() {
```
Comment (deepthidevaki) [inline zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ExportingEndpointIT.java:346]: :+1:

File: zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ExportingEndpointIT.java [hunk 5]
```diff
@@ -193,4 +206,181 @@ private void allPartitionsExporting() {
...
+
+    final var recordsAfterRestart =
+        Awaitility.await()
+            .atMost(Duration.ofSeconds(30))
+            .during(Duration.ofSeconds(5))
+            .until(RecordingExporter.getRecords()::size, hasStableValue());
+
+    // then
+    assertThat(recordsAfterRestart).isGreaterThan(recordsBeforePause);
+    Awaitility.await().untilAsserted(this::allPartitionsSoftPausedExporting);
```
Comment (deepthidevaki) [inline zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ExportingEndpointIT.java:384]: :x: Also verify the exporter position is not updated.

File: zeebe/protocol/src/main/resources/cluster-management-protocol.xml (hunk 2 above)
Comment (deepthidevaki) [reply zeebe/protocol/src/main/resources/cluster-management-protocol.xml:15]: :thinking: Not sure why this was not caught by the backward compatibility checker?

**Review by deepthidevaki — CHANGES_REQUESTED** [review]
:rocket: Looks good. Please take a look at the comments.

Some minor things to consider for future PRs.
- :upside_down_face: It would have been better to split this into to two PR.
- :upside_down_face: The PR description can be a bit more descriptive - for example by adding how the API endpoint would look like, how a user can send a request to this endpoint etc.

File: dist/src/main/java/io/camunda/zeebe/shared/management/ExportingEndpoint.java (hunk 1 above)
Comment (rodrigo-lourenco-lopes) [reply dist/src/main/java/io/camunda/zeebe/shared/management/ExportingEndpoint.java:24]: Having this as a separate endpoint helps with the testing in ExportingEndpointTest, and also makes the endpoints in line with the ones exposed on the broker. Do you think it is worth it to change here?

Comment (deepthidevaki) [reply dist/src/main/java/io/camunda/zeebe/shared/management/ExportingEndpoint.java:24]: You can still test it easily in ExportingEndpointTest. I don't see any problem.

There is no need to match the user facing API to match the internal implementation.

**Review by deepthidevaki — APPROVED** [review]
:rocket:  Please do some manual testing of this feature too.

**Pull request merged**